src/lib/loaders/loaders_with_authentication/vortex.ts

```diff
@@ -11,26 +19,60 @@ export default (accessToken, opts) => {
     vortexLoaderWithAuthenticationFactory,
   } = factories(opts)
 
+  const vortexAccessTokenLoader = () =>
+    vortexTokenLoader().then((data) => data.token)
+
   const gravityLoader = gravityLoaderWithAuthenticationFactory(
     gravityAccessTokenLoader
   )
 
+  const vortexLoader = vortexLoaderWithAuthenticationFactory(
+    vortexAccessTokenLoader
+  )
+
   // This generates a token with a lifetime of 1 minute, which should be plenty of time to fulfill a full query.
   const vortexTokenLoader = gravityLoader(
     "me/token",
     { client_application_id: VORTEX_APP_ID },
     { method: "POST" }
   )
 
-  const vortexAccessTokenLoader = () =>
-    vortexTokenLoader().then((data) => data.token)
+  const vortexGraphQLLoader = async <T = unknown>({
+    query,
+    variables,
+  }: GraphQLArgs): Promise<Record<string, T>> => {
+    const { token } = await vortexTokenLoader()
 
-  const vortexLoader = vortexLoaderWithAuthenticationFactory(
-    vortexAccessTokenLoader
-  )
+    const body = JSON.stringify({ query, variables })
+
+    const response = await fetch(urljoin(VORTEX_API_BASE, "graphql"), {
+      method: "POST",
+      body,
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    })
+
+    const json = await response.json()
+    const { data: vortexData, error, errors: vortexErrors } = json
+
+    if (error) {
+      throw new Error(`[loaders/vortex.ts] ${error.message}`)
+    } else if (vortexErrors) {
+      const errors = vortexErrors.reduce((acc, error) => {
+        return acc + " " + error["message"]
+      }, "[vortex]: Error ")
+
+      throw new GraphQLError(errors)
+    } else {
+      return vortexData
+    }
+  }
 
   return {
     vortexTokenLoader,
+    vortexGraphqlLoaderWithVariables: vortexGraphQLLoader,
```

Comment:
Have you tried using `vortexGraphqlLoader` and passing a query and variables like this:

```
vortexGraphqlLoader({
      query:  myQuery,
      variables: myVariables
    })()
    ```

Reply:
I actually tried but for some reason it raises issues when reading the variables